test/spec/test_spec.rb

```diff
@@ -0,0 +1,67 @@
+require_relative "spec_helper"
+
+describe "A PHP application" do
+
+  it "works with the getting started guide" do
+    Hatchet::Runner.new("php-getting-started").tap do |app|
+      app.deploy do
+      # deploy works
+      end
+    end
+  end
+
+  it "checks for bad version" do
+    Hatchet::Runner.new("php-getting-started", allow_failure: true).tap do |app|
+      app.before_deploy do
+        File.open("composer.json", "w+") do |f|
+          f.write '{
+          "require": {
+            "php": "7.badversion"
+          }}'
+        end
+      end
+      app.deploy do
+        expect(app.output).to include("Could not parse version constraint 7.badversion")
+      end
+    end
+  end
+
+  it "have absolute buildpack paths" do
+    buildpacks = [
+      :default,
+      "https://github.com/sharpstone/force_absolute_paths_buildpack"
+    ]
+    Hatchet::Runner.new("php-getting-started", buildpacks: buildpacks).deploy do |app|
+      #deploy works
+    end
+  end
+
+  it "uses cache with ci" do
+    app = new_app_with_stack_and_platrepo('test/fixtures/ci/devdeps')
+    app.run_ci do |test_run|
+      expect(test_run.output).to match("mockery/mockery")
+      expect(test_run.output).to include("Downloading")
+      test_run.run_again
+      expect(test_run.output).to include("Loading from cache")
+      expect(test_run.output).to_not include("Downloading")
+    end
+  end
+
+  it "should not restore cached directories when changing stack" do
```

Comment:
You want this test to fail, right? :)

The test actually succeeds, because the output **does** contain "Loading from cache" (see next paragraph). Since you're using `allow_failure` instead of explicitly asserting that the deploy output does **not** contain "Loading from cache", the test doesn't actually ensure anything - if the `expect()` succeeds, it passes, and if it does not, it passes, too. Same story as with the other `allow_failure` cases further above, except in this case, the problem is actually occurring!

The PHP buildpack does not have a stack-specific cache. Its cache will be re-used between stacks, and that is intentional, because the cache only contains userland PHP code, nothing compiled.

Reply:
> instead of explicitly asserting that the deploy output does not contain "Loading from cache"

Then the solution is to add a

```
      expect(app.output).to_not include("Loading from cache")
```

Before the stack update and deploy

> The PHP buildpack does not have a stack-specific cache. Its cache will be re-used between stacks, and that is intentional because the cache only contains userland PHP code, nothing compiled.

Right, so I think we want to keep the test and assert that the stack doesn't accidentally get cleared on stack upgrade.

Reply:
There was a misunderstanding over exactly what `allow_failure` does and doesn't do. We resolved it in slack